src/main/java/acme/roles/Sponsor.java

```diff
@@ -0,0 +1,40 @@
+
+package acme.roles;
+
+import javax.persistence.Entity;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+
+import org.hibernate.validator.constraints.Length;
+import org.hibernate.validator.constraints.URL;
+
+import acme.client.data.AbstractRole;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Entity
+public class Sponsor extends AbstractRole {
+
+	// Serialisation identifier -----------------------------------------------
+
+	private static final long	serialVersionUID	= 1L;
+
+	// Attributes -------------------------------------------------------------
+
+	@NotBlank
+	@Length(max = 75)
+	private String				name;
+
+	@NotBlank
+	@Length(max = 100)
+	private String				expectedBenefits;
+
+	@Email
```

Comment:
Add length constraints (between 0 and 255 according to on your tutorials, in url and email fields)
[https://ev.us.es/webapps/discussionboard/do/message?action=list_messages&course_id=_85092_1&nav=discussion_board_entry&conf_id=_405265_1&forum_id=_234042_1&message_id=_405561_1](url)